framework/src/media/InputHandler.cpp

```diff
@@ -345,6 +394,52 @@ ssize_t InputHandler::writeToStreamBuffer(unsigned char *buf, size_t size)
 	return (ssize_t)written;
 }
 
+audio_container_t InputHandler::getContainerFormat()
+{
+	mPreloadLength = 1024; // fixme
+
+	mPreloadData = new unsigned char[mPreloadLength];
```

Comment:
It is allocated when handler is opened, then should be freed when it closed, not only destructor.